### ISSUE TITLE
[IPv6] [Bare Metal] [VPS] Assign a /64 per machine

### ISSUE THREAD

**Issue by apio-sys** [ovh/infrastructure-roadmap#120]
(this is not a duplicate of #117 ).

Currently a VPS get's a (ridiculous) /128 assigned. Hence they are sitting in the same network (/64) as surrounding customers. So if you have a neighbour spamming using it's IPv6 he can get the whole /64 block blacklisted (which is normal practice, blacklisting should not occur below the /64 level) and you are taken down with him...

A VPS should get a /64 assignment and nothing lower.

As suggested in #117 , assigning a /56 or even /48 per NIC-handle would be great but will probably take a long time to implement. The real urgent problem here is not assigning /128 to any VPS anymore but only /64's. Address space is not an issue, there's more than enough of that (OVH having at least a /32 and a /31) and the service would already be a lot better.

Remaining at your disposal to discuss.

**Comment by LuckyWraptor** [comment]
Agreed, this should be implemented ASAP.

I'm not aware of how prefixes are delegated internally at OVH, and maybe this could require an entire infrastructural redesign.
But the fact is that however this is implemented, is just unacceptable, especially considering the fact that there is no firewall capabilities for IPv6 at all I don't see why they even thought this was a decent idea.

**Comment by apio-sys** [comment]
I'm glad to see this issue is getting some attention. Open to help or discuss further if needed but I reckon the scope is "easy". Hope this gets priority over #117 since will be much easier to implement IMHO and will not "waste" address space.

**Comment by eponier** [comment]
I'm interested in this too. As discussed in the first comment, IPv6 is not usable for the moment since there is necessarily a spammer in the same /64 as me.

**Comment by LuckyWraptor** [comment]
After nearly a full year, I got the issue resolved.

I switched provider

**Comment by apio-sys** [comment]
@thieummm @jslocinski is there any ETA for this ?

**Comment by apio-sys** [comment]
> After nearly a full year, I got the issue resolved.
> 
> I switched provider

That was probably the fastest solution indeed. A pity this isn't picked up more rapidly/seriuously. It has been much more than a year that this is an issue. @OVH Get your act together!

**Comment by jslocinski** [comment]
Renaming this to VPS only topic as baremetal servers are available with IPv6 /64 block by default.
We're working on more general, thus flexible solution (https://github.com/ovh/infrastructure-roadmap/issues/117), where /56 block will be available to our customers without permanent attachment to a particular infrastructure.

**Comment by apio-sys** [comment]
I thought it was already clear from the title and the contents that it was talking about VPS only (inside the Bare Metal menu from the admin perspective). I have noticed and previously referenced #117 and it would be great to implement that. But I can imagine that will not be very fast... Hence my intermediate solution to treat VPS as dedicated servers and just assign a /64 and the problem would already be solved in a much faster manner. We can't continue to work with a /128 per VPS. OVH has at least one /32 allocation which allows to distribute 4 billion /64 networks... The important question for me here is, what is the ETA to solve this on the VPS level ? #117 would be nice to have but is not mission critical. The issue at hand is.

**Comment by apio-sys** [comment]
@jslocinski the more flexible solution you mentioned above seems thus to be implemented using vRack hence not compatible with bare metal VPS's. This ticket has been acknowledged, so how do you plan to solve it? Can't you just assign a /64 and this will be "solved". This has been dragging on for years now...

**Comment by Gerard-Lh** [comment]
Hello everyone. I insist that each VPS should receive a /64 block, not a /128. Currently, in our cPanel environment, we are experiencing IPv6 address issues because sharing the same /64 means that one user's misuse affects everyone. Additionally, for high-performance VPS like the VLE, having a /64 is especially necessary. Please fix this as soon as possible!

**Comment by apio-sys** [comment]
Totally agree @Gerard-Lh ! It's a question of respecting industry standards. One should never assign /128's to machines that don't even "know each other" (different customers end up in the same network!). Heck, you wouldn't even do this in your home network. Address space is not an issue and one should not try to "save" IPv6 address space. We have more than enough of it for decades to come.

That was my yearly "contribution" to this. Hopefully we will get some response finally beyond the vRack solution that does not apply to VPS's. @jslocinski @thieummm any comments from OVHCloud on this? If it's a "wontfix" case for you guys, then please be so honest to share that so that all that seeing this as a menace to their production can take appropriate steps and migrate their services elsewhere. Would be a shame though for a rather simple issue. Maybe ask Octave's opinion also.

**Comment by yverry** [comment]
listed this morning in spamhaus
> 2001:41d0:304:200::/64 initiated a tcp connection from 2001:41d0:304:200::/64 using source port 46226, to the sinkhole IP address unknown on destination port 80.
>
>The most recent detection was on: April 5 2025, 08:17:38 UTC.

I've raised a trouble ticket but I think this is not useful has expected

**Comment by apio-sys** [comment]
Indeed @yverry, that is not useful since Spamhaus will block a /64 network which is misbehaving. If you have a machine assigned a poor /128 in that network and one of the (very many) neighbours you will have there misbehaves, you will go down with them. This forces you to prefer sending over IPv4 which is obviously not going towards the future. There should NEVER be a /128 assignment to a machine.

![Image](https://github.com/user-attachments/assets/b8452b76-1a44-4509-b3b3-a8d95fc2eb89)

@ovhcloud @mhurtrel @jslocinski @octaveklaba @olesovhcom.bsky.social please help finally solving this on the OVH end.

**Comment by Kurgan-** [comment]
I second this request because of blacklist issues.

**Comment by rob4226** [comment]
With blacklist issues, sometimes even a /64 is not good enough! Really a server (baremetal or VPS) should have a /48 assigned to it.  At the very least, a /64 should be assigned.  With a /128 you might as well disable IPv6 if you have an email server.

**Comment by apio-sys** [comment]
/64 for a server is enough and good practice. You won't get blacklisted for a neighboring /64 since it is your own network. And indeed just a /128 you are inclined to disable IPv6 especially for an email server. And we want people to embrace IPv6 and not disabling it... @jslocinski @martinovh @jpatovh is this finally fixed in the VPS2026 product range?

**Comment by ramonmedina-net** [comment]
I recently signed up for a VPS and discovered that it's in the blacklisted IPv6 /64 range. 

Please fix this!

**Comment by avidality** [comment]
@ramonmedina-net 
same here with one year commitment.
 
/64 ipv6 is blacklisted @ spamhaus
ipv4 is on UCEPROTECT-Level3 (at the ASN level) and also on UCEPROTECT-Level2 almost every other week.

not sure if and how I can use vRack for VPS.

**Comment by apio-sys** [comment]
@avidality unfortunately you can't use vRack with VPS. Only with Dedicated servers and Public Cloud.

**Comment by ramonmedina-net** [comment]
This is the response I received from Spamhaus. The IP they reference is NOT the one assigned to me, but is in the same /64 range.

`Thank you for contacting the Spamhaus Ticketing team,

Is this exact IP yours? Do you control the full /64 or larger?
------------------------------------------
IP: 2604:2dc0:101:200::3569
------------------------------------------

If the answer is no, please contact your service provider.

CSS lists IPv6 with /64 granularity, and one or more IPs within the /64 in question are sending spam.

A /64 is the industry standard for the smallest IPv6 allocation to individual customers, even for home-uses like cable, DSL or wireless. The /64 choice has RFC4291 as its origin and it is further discussed in RFC6177.

As a spammer can cycle through IPv6 addresses every 5 seconds or less and an IPv6 /64 is twice the size of the entire IPv4 internet, a lot of spam can be sent with minimal effort. Thus, listing the whole /64 is warranted.

For more detailed information, the below URL should help:

https://www.spamhaus.org/faqs/combined-spam-sources-css/#css-how-handle-ipv6

To resolve this issue, contact your service provider and request the assignment of a /64 range together with DNS/rDNS appropriate for email service.

These links may be helpful when having the conversation with your ISP:

https://slash64.net/

https://www.m3aawg.org/sites/default/files/document/M3AAWG_Inbound_IPv6_Policy_Issues-2014-09.pdf



Regards,
R e`

**Comment by apio-sys** [comment]
Yes @ramonmedina-net , hence my ticket but @OVH is unfortunately not paying much attention since I posted it now 3 years ago already to it a part from the fact that they acknowledged it.

**Comment by Stephane-Bautista** [comment]
Hi everyone,

Thank you all for your patience and for continuing to share your experiences and concerns around IPv6 allocation on VPS.

We completely understand the impact of the current /128 setup - especially regarding blacklisting and isolation issues when multiple customers share the same /64. This feedback is absolutely valid and well understood internally.

That said, moving from a shared /64 model to assigning a dedicated /64 per VPS is **not a small change**. It requires a **fundamental redesign of how our network is structured and delivered** across the VPS infrastructure. Our teams are actively investigating the best way to implement this at scale.

Because of this complexity, we need to carefully plan and validate the approach before we can commit to a deployment timeline. But please rest assured: the topic is well identified, being discussed at a high technical level, and remains on our roadmap.

We’ll post an update here as soon as we have clearer progress or decisions to share.

Thank you again for your understanding and for helping us improve our platform.

Stéphane

**Comment by apio-sys** [comment]
Thanks @Stephane-Bautista for this promising update and glad to hear you are taking the topic serious. This can be a showstopper for lots of your customers, especially smaller ones.

**Comment by nlgranger** [comment]
@Stephane-Bautista as a stop-gap, maybe consider keeping the /128 networking but edit whatever assigns addresses to use steps of 2**64 so that each vps ends up in a separate /64 subnet.

I suspect this would be a much smaller change to implement and would resolve the urgent matter of blacklists.

**Comment by axiades** [comment]
Please Upgrade the Kimsufi-Servers to an IPv6 /112 or an Full /64 Block cause there are over a Million of IP-Adresses that are not Usable with the Kimsufi Servers if they are Limited to an /128 of an Full /64 net

**Comment by K0n24d** [comment]
I just don't get it, the beginning of IPv6 deployment was over 20 years ago. Yet in the "NEW VPS 2026" the way IPv6 was implemented is completely flawed: /128 assignment, no IPv6 failover...
By the way, knowing nothing about your internal infrastructure, but IPv6 failover might me easier to implement.

**Comment by nickperkins** [comment]
/128 assignment is criminal. I am going to have to move my mail servers to another provider.

**Comment by Stephane-Bautista** [comment]
Hello everyone,

I wanted to give you a quick update on this topic. As I mentioned earlier, we are taking the **RFC 6177** compliance very seriously, and our engineering teams are working hard on the long-term architecture.

Since the full redesign of our network orchestration takes time to implement at scale, we don’t want to leave you waiting without any improvements. Therefore, we are launching a **Quick Win within the coming months** to address the most urgent pain point: **IP Reputation & Isolation**.

What’s coming:
We are moving away from the shared /64 model. Each VPS will soon be assigned to its own **isolated /64 segment**.
- **The benefit**: You will no longer share your prefix with other customers, effectively eliminating "noisy neighbor" risks (blacklisting, reputation issues, etc.).
- **The catch**: In this first phase, usage will still be limited to a single **/128 address** within that segment.

I’ve created a dedicated task to track the deployment of this Quick Win here ==> https://github.com/ovh/infrastructure-roadmap/issues/456

Stephane

**Comment by apio-sys** [comment]
Thanks for that promising move forward @Stephane-Bautista . I have read your update in #456 but would like to know what and when are the next steps planned since there is no update since March 12. I have just tested an order for a fresh VPS. In `/etc/netplan/50-cloud-init.yaml` it gets à /128 assigned (by the way, you should review your netplan syntax since on a fresh delivery it has deprecated syntax and wrong permissions causing errors). Is this now isolated inside a /64? I think still not. Since I was blacklisted for that /64 at Spamhaus as before. Looking forward to some news and a proof of concept and how we - as customers or resellers - can check and proof this.

**Comment by sbraz** [comment]
> by the way, you should review your netplan syntax since on a fresh delivery it has deprecated syntax and wrong permissions causing errors

Hello, that's cloud-init creating the file. Not sure what distribution you're on but this should be fixed in recent cloud-init releases. That's not something we at OVHcloud can control.

EDIT with additional context:

We rely on cloud-init for Netplan configuration generation. The two bugs you mention are fixed respectively in https://github.com/canonical/cloud-init/commit/076b51b362402d821414ee329a90c6ecd5b8aa7a and https://github.com/canonical/cloud-init/commit/4e08e70e5c4201c55a88f94529d7c222ba4f2277. For some reason, they do not seem to be included in Debian 13's [25.1.4](https://packages.debian.org/trixie/cloud-init) cloud-init version.

If other providers do not have this issue, it may be because their Debian images rely on ifupdown (`/etc/network/interfaces`) instead. Or they do not use cloud-init for provisioning at all

Our policy is to modify cloud images as little as possible. We create our Debian 13 images from https://cdimage.debian.org/cdimage/cloud/trixie/latest/ and these "source" images are affected by the two issues you mentioned. Patching cloud-init to change the way it generates Netplan config files is non-trivial and touching Netplan after cloud-init runs would be tricky too.

**Comment by apio-sys** [comment]
> Hello, that's cloud-init creating the file. Not sure what distribution you're on but this should be fixed in recent cloud-init releases. That's not something we at OVHcloud can control.

This example is from Debian 13. I think you have perfect control over this since you provide the images for your clients no? It works with other providers so nothing to do with the distro itself. You have the following errors:

```
** (process:203988): WARNING **: 18:39:40.105: Permissions for /etc/netplan/50-cloud-init.yaml are too open. Netplan configuration should NOT be accessible by others.
```

easy to fix with a `chmod 0600` but this should be done before delivery. And:

```
** (process:203988): WARNING **: 18:39:40.106: gateway6 has been deprecated, use default routes instead.
See the 'Default routes' section of the documentation for more details
```

That is not so good and needs to be properly rewritten with `routes to via`, etc. Anyway, that is not the subject of this ticket here so let's not pollute it with other info and concentrate on the proper IPv6 assignments.

_EDIT 2026-04-20: this must have been on an older VPS. I just rechecked it after a fresh install and the above 2 issues are corrected. So let's get back to the original subject._

**Comment by apio-sys** [comment]
> I have just tested an order for a fresh VPS. In `/etc/netplan/50-cloud-init.yaml` it gets à /128 assigned. Is this now isolated inside a /64? I think still not. Since I was blacklisted for that /64 at Spamhaus as before. Looking forward to some news and a proof of concept and how we - as customers or resellers - can check and proof this.

Follow-up on my own question: it is not in place. I opened a ticket with spamhaus to check why my new IP 2001:41d0:305:2100::7209 is blacklisted and I had manual confirmation that the scatter came from 2001:41d0:305:2100::88c3 within the same /64. 

Looking forward for the next constructive feedback asap.

**Comment by apio-sys** [comment]
Time for a quarterly PING? Can you provide ANY update at all after announcing a quick win in March and no feedback since?

**Comment by Stephane-Bautista** [comment]
Hello @apio-sys,

First, please accept my apologies for the late reply.
I wanted to let you know that the Quick Win is now fully live in production. Every new VPS is now provisioned within its own isolated /64 segment to protect your IP reputation. Please note that during this first phase, your OS will still display a /128 address, but the underlying infrastructure routing strictly isolates your /64.

Thanks again for your valuable feedback and for helping us improve the service!

Best regards,
Stephane

**Comment by kav2k** [comment]
Is there a way to apply this to existing VPSes?

**Comment by apio-sys** [comment]
Thanks for the update @Stephane-Bautista . The next logical questions are:

1. How can we verify it is applied to a specific VPS or not? Since when is this the case? I have a recent order from April but can't see a difference if it is in the new case or not.
2. How to apply to existing machines? Would be nice to propose an IPv6 change for those that want to to apply the new system (through API or a support ticket?).

Looking forward to the next constructive updates. We are getting close to something finally usable provided we change existing machines.

**Comment by Pieshka** [comment]
I’m interested in the same thing. I’ve had a VPS 2026 with an IPv6 address for some time now, but I don’t use that address because it belongs to a shared /64 subnet and has a bad reputation. Contacting OVH technical support didn’t help much, and I didn’t find out from them whether Quick Win is available for existing VPSs. I think the ability to implement this on existing services would be sufficient for many people, and it would be great if that were possible.

**Comment by debtquity** [comment]
@Stephane-Bautista:

> … I wanted to let you know that the Quick Win is now fully live in production. Every new VPS is now provisioned within its own isolated /64 segment to protect your IP reputation. Please note that during this first phase, your OS will still display a /128 address, but the underlying infrastructure routing strictly isolates your /64.

Seems there’s a disconnect between what is communicated here and what exists available to customers. 

I checked with support group to confirm if my VPS tagged with `VPS-2 2027` was included in this "quick win". Per "US VPS Product Management team", this initiative was rolled back with plans to implement in future but no planned delivery date.

**Comment by edthepurple** [comment]
Hello. I don't know if I'm commenting in somewhere I shouldn't or not - I'm just a customer. may I know when are we getting /64 subnets on our vps? I'm a few clicks away from moving all my servers from Hetzner to OVH.

**Comment by voidzero** [comment]
Extremely frustrating that this problem has not been fixed by OVH. A single IPv6 address is just not how to go about IPv6. Give every server a /64 and/or every customer a /48. But Spamhaus is just not accepting single IPv6 addresses and this should be done differently by OVHCloud. If it's an anti-abuse measure, this is not the right way to go about doing this. Instead, treat /64 like a single IPv4. Now I can't use IPv6 at all because Spamhaus is blocking the /64 of which I have a single address. They are doing it right - OVHCloud is not. Please fix ASAP.

**Comment by Stephane-Bautista** [comment]
Hello everyone,

I have posted a detailed update regarding the status of this Quick Win, the rollback, and what’s next on the main issue here: https://github.com/ovh/infrastructure-roadmap/issues/456

Best regards,
Stephane

**Comment by apio-sys** [comment]
So almost 4 years after I opened this ticket and à part from acknowledgement of the problem we are nowhere with this. Quite unbelievable. If OVH doesn't want to take this into consideration (anymore) then I'll leave you close this as wontfix. I guess some of us will have to order VPS's from other providers, what a pity.
